Set CI and Scoring Method based on Data

JanRose1 opened the pull request:
### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **dev branch** (left side). Also you should start *your branch* off *our dev*.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.

### Issue number(s) that this pull request fixes
- Fixes #1873

### List of changes to the codebase in this pull request
- Added function to PC, GES, and HillClimb estimators that checks input data to set proper Scoring Function and CI_Test
-
-

JanRose1 committed:
Set CI_Score based on data

Added a function to the files that will set the proper CI_Test or Scoring function based on the input data (Continuous, Categorical, or Mixed variable).

JanRose1 committed:
Merge pull request #1 from JanRose1/JanRose1-Set_CI_Score

Set CI_Score based on data

JanRose1 committed:
Added CI_Test Check

JanRose1 committed:
Merge pull request #2 from JanRose1/JanRose1-CI_Test_Check_PC

Added CI_Test Check

ankurankan commented:
@JanRose1 Thanks for the PR. GitHub actions are failing because of code formatting. Could you run `black <filename>` on the files you have made changes to? For future changes, you can also install a pre-commit hook by running `pre-commit install` from the pgmpy dir to automatically run these checks when creating commits.

ankurankan commented on pgmpy/estimators/GES.py:
```diff
@@ -55,6 +55,20 @@ def __init__(self, data, use_cache=True, **kwargs):
 
         super(GES, self).__init__(data=data, **kwargs)
 
+    def Test_type(self):#Ideally I would run this in over to choose the proper test depending on the kind of data.
```
1. We use snake case in pgmpy, so you could name the function as `test_type`. 
2. I think it would be nice to have a more informative name something such as `get_scoring_method`
3. I think it would be better to move the method to `StructureScore.py`. We can then import the method in all the algorithm files. This would avoid duplicated code.

ankurankan replied:
I would also suggest removing the comment here and writing a docstring instead. You can find examples of docstring format in most of the other methods.

ankurankan commented on pgmpy/estimators/PC.py:
```diff
@@ -50,10 +50,25 @@ class PC(StructureEstimator):
     def __init__(self, data=None, independencies=None, **kwargs):
         super(PC, self).__init__(data=data, independencies=independencies, **kwargs)
 
+    def Test_type(self):
```
Similarly, I would suggest moving this to `CITests.py`

JanRose1 committed:
CI_Tests and Scoring Reducing Redundancy

Added function for testing input data to StuctureScore.py and CITests.py to reduce code redundancy. Modified GES, HillClimbSearch, and PC.py to import the function from StructureScore.py or CITests.py as needed. Modified __init__.py to import scoring class from StructureScore.py.

JanRose1 committed:
Ran Black

Ran Black on the files so they'd be formatted properly.

JanRose1 commented:
Ok so I ran black on the files I modified though maybe this introduced some errors elsewhere? Yet I'm unsure why some checks passed with others being cancelled.

ankurankan commented:
@JanRose1 The test are split onto separate machines to run them parallelly. And if any of them fails, it cancels all the other checks. You can look at the tests that failed here: https://github.com/pgmpy/pgmpy/actions/runs/12208977751/job/34063070262?pr=1878#step:8:449. I think it's failing because the default scoring method has changed now resulting in a different learned structure.

JanRose1 commented:
Oh I see, so would the checks have to be changed accordingly or is this a matter of making sure the learned structure leads to the correct output? If it’s the latter I can def check it out later. Definitely let me know if you have any other tips or suggestions.Sent from my iPhoneOn Dec 7, 2024, at 10:53 AM, Ankur Ankan ***@***.***> wrote:﻿
@JanRose1 The checks split the tests onto separate machines to run them parallel. And if any of them fails, it cancels all the other machines. You can look at the tests that failed here: https://github.com/pgmpy/pgmpy/actions/runs/12208977751/job/34063070262?pr=1878#step:8:449. I think it's failing because the default scoring method has changed now resulting in a different learned structure.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

ankurankan commented:
@JanRose1 You can either change the current test to check for the new structure with the default scoring methods. Or you can specify the old default scoring method in the tests. Or you test both of these scenarios. We ideally want the test coverage to be as high as possible (i.e., our test should run all possible scenarios).

JanRose1 commented:
Sounds good 👍 will get on it for both.Sent from my iPhoneOn Dec 7, 2024, at 11:30 AM, Ankur Ankan ***@***.***> wrote:﻿
@JanRose1 You can either change the current test to check for the new structure with the default scoring methods. Or you can specify the old default scoring method in the tests. Or you test both of these scenarios. We ideally want the test coverage to be as high as possible (i.e., our test should run all possible scenarios).

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

JanRose1 commented:
Hey sorry for the wait. I meant to get on this a bit sooner but a personal emergency came up and has taken over my time. I hope to finish tackling the unit tests before the end of this week. Thank you for your patience.

ankurankan commented:
@JanRose1 No worries, take your time. And let me know if you need any help :)

ankurankan commented:
`get_scoring_method` and `get_ci_test` methods have been implemented. Closing this.
